appengine/src/main/java/com/google/googleinterns/gscribe/services/impl/SpreadsheetServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+     * If sheet already exists then clear the sheet
+     * If sheet does not exist then create new sheet with name sheetName in the spreadsheet
+     *
+     * @param service       ( sheet service object )
+     * @param spreadsheetId ( spreadsheet id of spreadsheet where question paper lies )
+     * @param sheetName     ( sheet name of sheet which is to be created or cleared )
+     * @throws IOException ( if any error occurs regarding access of spreadsheet )
+     */
+    private void sheetSetup(Sheets service, String spreadsheetId, String sheetName) throws IOException {
+
+        Sheets.Spreadsheets.Get spreadsheetService = service.spreadsheets().get(spreadsheetId);
+        Spreadsheet spreadsheet = spreadsheetService.execute();
+        List<Sheet> sheets = spreadsheet.getSheets();
+        boolean sheetAlreadyMade = false;
+        for (Sheet sheet : sheets) {
+            if (sheet.getProperties().getTitle().equals(sheetName)) {
+                ClearValuesRequest requestBody = new ClearValuesRequest();
+                Sheets.Spreadsheets.Values.Clear request = service.spreadsheets().values().clear(spreadsheetId, sheetName, requestBody);
+                request.execute();
+                sheetAlreadyMade = true;
+            }
+        }
+        if (!sheetAlreadyMade) {
+            AddSheetRequest addSheetRequest = new AddSheetRequest();
+            SheetProperties sheetProperties = new SheetProperties();
+            sheetProperties.setTitle(sheetName);
+            addSheetRequest.setProperties(sheetProperties);
+            List<Request> requestsList = new ArrayList<>();
+            requestsList.add(new Request().setAddSheet(addSheetRequest));
+            BatchUpdateSpreadsheetRequest batchUpdateSpreadsheetRequest = new BatchUpdateSpreadsheetRequest();
+            batchUpdateSpreadsheetRequest.setRequests(requestsList);
+            service.spreadsheets().batchUpdate(spreadsheetId, batchUpdateSpreadsheetRequest).execute();
+        }
+
+    }
+
+    /**
+     * Called to create responses sheet for the exam
+     * Sets up the responses sheet using sheetSetup function
+     * Forms header for the response sheet as ( Start time, roll number, questions ..., final points )
+     * Add header to the sheet
+     *
+     * @param exam ( exam object to be filled into the spreadsheet )
+     * @param user ( user object for tokens to access spreadsheet )
+     * @throws GeneralSecurityException,IOException ( thrown by NetHttpTransport, GoogleClientSecrets, GoogleTokenResponse or by invalid credentials file  )
+     */
+    private void fillResponseTemplate(Exam exam, User user) throws GeneralSecurityException, IOException {
+        /* Set access token to get the spreadsheet Instance */
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
```

Comment:
Too much code duplication.

Reply:
Injected the fields and created a private variable for class to avoid duplication of code.